contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol

```diff
@@ -464,11 +435,15 @@ contract RibbonTreasuryVault is
      * @notice Deposits the `asset` from msg.sender.
      * @param amount is the amount of `asset` to deposit
      */
-    function deposit(uint256 amount) external onlyWhitelist nonReentrant {
+    function deposit(uint256 amount) external nonReentrant {
         require(amount > 0, "!amount");
 
         _depositFor(amount, msg.sender);
 
+        // if (!depositorsMap[msg.sender]) {
+        _addDepositor(msg.sender);
```

Comment:
We can move this to before `_depositFor` so that it throws when they exceed the depositor limit